matsim/src/main/java/ch/sbb/matsim/routing/pt/raptor/RaptorRoute.java

```diff
@@ -75,6 +75,12 @@ void addPt(TransitStopFacility fromStop, TransitStopFacility toStop, TransitLine
         this.ptLegCount++;
     }
 
+	void addChainedPart(TransitStopFacility fromStop, TransitStopFacility toStop, TransitLine line, TransitRoute route, String mode, double depTime, double boardingTime, double vehDepTime, double arrivalTime, double distance) {
+		RoutePart part = new RoutePart(fromStop, toStop, mode, depTime, boardingTime, vehDepTime, arrivalTime, distance, line, route, null);
+		this.travelTime += (arrivalTime - depTime);
+		editableParts.getLast().chainedPart = part;
```

Comment:
What if we have 3 or more departures chained, e.g. in circular routes? Shouldn't the new part be attached to the  end of already chained parts? something like:

```java
RoutePart lastPart = editableParts.getLast();
while (lastPart.chainedPart != null) {
  lastPart = lastPart.chainedPart;
}
lastPart.chainedPart = part;
```